smalltalksrc/Slang-Tests/SlangBasicTranslationTestClass.class.st

```diff
@@ -195,6 +195,15 @@ SlangBasicTranslationTestClass >> methodWithAndIntegerReceiver [
 	result := receiver & 1
 ]
 
+{ #category : #'as yet unclassified' }
+SlangBasicTranslationTestClass >> methodWithAndIntegerReceiverAndBooleanArgument [
+
+	| receiver result |
+	
+	receiver := 1000.
+	result := receiver & false
```

Comment:
What if the boolean is an argument?

```smalltalk
methodWithAndIntegerReceiverAndBooleanArgument: arg [

	| receiver result |

	receiver := 1000.
	result := receiver & arg
```

and all variantas :)

- boolean & boolean
- boolean & int
- int & boolean
- int & int

Reply:
The cases where both are boolean/int are covered with `methodWithAndBooleanReceiver` and `methodWithAndIntegerReceiver`. Maybe the name should reference the argument 😅  

The other cases I considered them all the same test case (covered by this method), as they fall under the "if they are different types then throw an error" logic.

Reply:
Yes, but my point is that those tests are on literal booleans/integers!!!
What if the receiver are the result of a more complex expression, say a variable? Do we need a type annotation for booleans?